app/lib/package/backend.dart

```diff
@@ -377,14 +377,27 @@ class GCloudPackageRepository extends pub_server.PackageRepository {
 
   GCloudPackageRepository(this.db, this.storage);
 
-  // Metadata support.
-
-  @override
-  Stream<pub_server.PackageVersion> versions(String package) {
-    final packageKey = db.emptyKey.append(Package, id: package);
-    final query = db.query<PackageVersion>(ancestorKey: packageKey);
-    return query.run().map((model) => pub_server.PackageVersion(
-        package, model.version, model.pubspec.jsonString));
+  /// Returns the known versions of [package].
+  Future<api.PackageData> packageData(Uri baseUri, String package) async {
```

Comment:
Can we call this `listVersions`?

Reply:
Ideally, also a small doc comment that this is what we use in `pub` for finding which versions exist.